tests/Proto.TestFixtures/TestMailboxHandler.cs

```diff
@@ -2,11 +2,16 @@
 using System.Collections.Generic;
 using System.Threading.Tasks;
 using Proto.Mailbox;
+using System.Threading;
+using System.Collections.Concurrent;
 
 namespace Proto.TestFixtures
 {
     public class TestMailboxHandler : IMessageInvoker, IDispatcher
     {
+        private ConcurrentQueue<TaskCompletionSource<int>> _taskCompletionQueue =
```

Comment:
This allows for better parity between Schedule() and ResumeMailboxProcessingAndWaitAsync in case ResumeMailboxProcessingAndWaitAsync needs to be called more than once.